src/modules/emotes/channel-emotes.js

```diff
@@ -12,43 +12,46 @@ const category = {
   displayName: 'BetterTTV Channel Emotes',
 };
 
+let currentChannel = null;
+
 class ChannelEmotes extends AbstractEmotes {
   constructor() {
     super();
-
     watcher.on('channel.updated', (d) => this.updateChannelEmotes(d));
   }
 
   get category() {
     return category;
   }
 
+  upsertChannelEmote({id, user, imageType, code}) {
+    this.emotes.set(
+      code,
+      new Emote({
+        id,
+        category: this.category,
+        channel: user || currentChannel,
```

Comment:
reviewing this logic again, since we are calling `setCurrentChannel` we could leverage `getCurrentChannel` here instead of storing this state in the current file too

Reply:
do we not need the avatar stored here?